src/Http/Http/test/ResponseCookiesTest.cs

```diff
@@ -74,6 +75,37 @@ public void DeleteCookieShouldSetDefaultPath()
             Assert.Contains("expires=Thu, 01 Jan 1970 00:00:00 GMT", cookieHeaderValues[0]);
         }
 
+        [Fact]
+        public void DeleteCookieWithDomainAndPathDeletesPriorMatchingCookies()
```

Comment:
Note this new test isn't exercising the product fix you added. E.g. if you comment out your fix this test will still pass.

A closer test is NoParamsDeleteRemovesCookieCreatedByAdd where it first adds a cookie and then deletes it, that's where the filter logic gets used.

Reply:
Wait, I see that the filter is partially used against the prior delete cookies. You're still missing cases where the filter actually removes a prior matching cookie though, right?

Reply:
Right 🙂 . We can add a test, with the same behavior **`NoParamsDeleteRemovesCookieCreatedByAdd`** test, but with path and domain specified.